src/data/equiv/basic.lean

```diff
@@ -1725,6 +1725,15 @@ protected noncomputable def image_of_inj_on {α β} (f : α → β) (s : set α)
 protected noncomputable def image {α β} (f : α → β) (s : set α) (H : injective f) : s ≃ (f '' s) :=
 equiv.set.image_of_inj_on f s (H.inj_on s)
 
+@[simp] protected lemma image_symm_apply {α β} (f : α → β) (s : set α) (H : injective f)
+  (x : α) (h : x ∈ s) :
+  (set.image f s H).symm ⟨f x, ⟨x, ⟨h, rfl⟩⟩⟩ = ⟨x, h⟩ :=
```

Comment:
Somehow I find it easier to read when angle bracket expressions are flattened out, but this is a very weak suggestion.
```suggestion
  (set.image f s H).symm ⟨f x, x, h, rfl⟩ = ⟨x, h⟩ :=
```
By the way, in the proof below if you remove `only` the first `simp` proves the goal.

Reply:
I prefer the unflattened one, I guess.